File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    for i in range(len(discount)-10+1):
        v = [0 for _ in range(len(want))]
        for j in range(len(want)):
            if discount[i:i+10].count(want[j])>=number[j]:
                v[j] = 1
        if all(v):
            answer += 1
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        int n = discount.length - 10;
        
        // 원하는 제품 - 갯수 같이 저장
        HashMap<String, Integer> wMap = new HashMap<>();
        for (int i=0; i<want.length; i++){
            wMap.put(want[i], number[i]);
        }
        
        for (int i=0; i<=n; i++){
            HashMap<String, Integer> tMap = new HashMap<>(wMap);
            boolean flag = true;
            // 연속 10일 간 원하는 제품이 일치하는 경우만 카운트
            for(int j=i; j<i+10; j++){
                if(!tMap.containsKey(discount[j])){
                    flag = false; 
                    break;
                }
                int cnt = tMap.get(discount[j])-1;
                if(cnt==0) tMap.remove(discount[j]);
                else tMap.put(discount[j], cnt);
            }
            if(flag) answer ++;
        }
        
        return answer;
    }
}